src/app/state/slices/api/gameboards.ts

```diff
@@ -48,7 +48,7 @@ export const assignGameboard = createAsyncThunk(
 
     // TODO think about whether this can be done in the back-end too?
     if (dueDate !== undefined) {
-      dueDate?.setHours(0, 0, 0, 0);
+      new Date(dueDate).setHours(0, 0, 0, 0);
```

Comment:
You're not updating `dueDate` here. You're essentially creating a new object, updating it, and then never using it again, as it's not assigned to anything.